nullaway/src/main/java/com/uber/nullaway/fixserialization/Location.java

```diff
@@ -0,0 +1,143 @@
+ * THE SOFTWARE.
+ */
+
+package com.uber.nullaway.fixserialization;
+
+import com.google.common.base.Preconditions;
+import com.google.errorprone.util.ASTHelpers;
+import com.sun.tools.javac.code.Symbol;
+import java.net.URI;
+import javax.lang.model.element.ElementKind;
+
+/** Holds all the information to target a specific element in the source code. */
+public class Location implements SeparatedValueDisplay {
+
+  private final URI uri;
+  private final Symbol.ClassSymbol enclosingClass;
+  private Symbol.MethodSymbol enclosingMethod;
+  /** Symbol of the element if the target is either a class field or a method parameter. */
+  private Symbol.VarSymbol variableSymbol;
+
+  private Kind kind;
+  /** Index of the element if target is a method argument. */
+  int index = 0;
+
+  public enum Kind {
+    CLASS_FIELD("CLASS_FIELD"),
+    METHOD_PARAM("METHOD_PARAM"),
+    METHOD_RETURN("METHOD_RETURN");
+    public final String label;
+
+    Kind(String label) {
+      this.label = label;
+    }
+  }
+
+  /**
+   * Initializes properties based on the type of the target.
+   *
+   * @param target Target element.
+   */
+  public Location(Symbol target) {
+    switch (target.getKind()) {
+      case PARAMETER:
+        createMethodParamLocation(target);
+        break;
+      case METHOD:
+        createMethodLocation(target);
+        break;
+      case FIELD:
+        createFieldLocation(target);
+        break;
+      default:
+        throw new IllegalArgumentException("Cannot locate node: " + target);
+    }
+    this.enclosingClass = ASTHelpers.enclosingClass(target);
+    this.uri = enclosingClass.sourcefile.toUri();
+  }
+
+  private void createFieldLocation(Symbol field) {
+    Preconditions.checkArgument(field.getKind() == ElementKind.FIELD);
+    variableSymbol = (Symbol.VarSymbol) field;
+    kind = Kind.CLASS_FIELD;
+  }
+
+  private void createMethodLocation(Symbol method) {
+    Preconditions.checkArgument(method.getKind() == ElementKind.METHOD);
+    kind = Kind.METHOD_RETURN;
+    enclosingMethod = (Symbol.MethodSymbol) method;
+  }
+
+  private void createMethodParamLocation(Symbol parameter) {
+    Preconditions.checkArgument(parameter.getKind() == ElementKind.PARAMETER);
+    this.kind = Kind.METHOD_PARAM;
+    this.variableSymbol = (Symbol.VarSymbol) parameter;
+    Symbol enclosingMethod = parameter;
+    // Look for the enclosing method.
+    while (enclosingMethod != null && enclosingMethod.getKind() != ElementKind.METHOD) {
+      enclosingMethod = enclosingMethod.owner;
+    }
+    Preconditions.checkNotNull(enclosingMethod);
+    this.enclosingMethod = (Symbol.MethodSymbol) enclosingMethod;
+    for (int i = 0; i < this.enclosingMethod.getParameters().size(); i++) {
+      if (this.enclosingMethod.getParameters().get(i).equals(parameter)) {
+        index = i;
+        break;
+      }
+    }
+  }
+
+  @Override
+  public String display(String delimiter) {
```

Comment:
This doesn't display anything.  Maybe `toStringWithDelimiter()`?

Reply:
031bae163089e43b14107937de2460c8ce4d8664